### PR TITLE
Fix interpreter tests with numpy-2.0

### LINKED ISSUE #2058: Interpreter tests failures with numpy-2.0.2
```
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int8-32] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer 351 out of bounds for int8')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int8-64] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -334 out of bounds for int8')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int8-128] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -1387 out of bounds for int8')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int8-512] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -3350 out of bounds for int8')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int16-32] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer 129477 out of bounds for int16')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int16-64] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer 101665 out of bounds for int16')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int16-128] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -83455 out of bounds for int16')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int16-512] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer 517704 out of bounds for int16')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int32-32] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -4736833654 out of bounds for int32')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int32-64] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -17340316748 out of bounds for int32')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int32-128] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer -15950412024 out of bounds for int32')")
FAILED language/test_core.py::test_reduce1d[4-1-1-sum-int32-512] - triton.runtime.errors.InterpreterError: InterpreterError("OverflowError('Python integer 6080852592 out of bounds for int32')")
```

https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10621771273/job/29444606009#step:20:18210

### PR DESCRIPTION
**Pull request by Retribution98**
Closes #2058 

The core Triton is a small number of people, and we receive many PRs (thank
you!).  To help us review your code more quickly, **if you are a new
contributor (less than 3 PRs merged) we ask that you complete the following
tasks and include the filled-out checklist in your PR description.**

Complete the following tasks before sending your PR, and replace `[ ]` with
`[x]` to indicate you have done them.

- [x] I am not making a trivial change, such as fixing a typo in a comment.

- [x] I have written a PR description following these
  [rules](https://cbea.ms/git-commit/#why-not-how).

- [x] I have run `pre-commit run --from-ref origin/main --to-ref HEAD`.

- Select one of the following.
  - [ ] I have added tests.
    - `/test` for `lit` tests
    - `/unittest` for C++ tests
    - `/python/test` for end-to-end tests
  - [x] This PR does not need a test because `FILL THIS IN`.

- Select one of the following.
  - [x] I have not added any `lit` tests.
  - [ ] The `lit` tests I have added follow these [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices),
    including the "tests should be minimal" section. (Usually running Python code
    and using the instructions it generates is not minimal.)

### CODE REVIEW

File: python/triton/runtime/interpreter.py [hunk 1]
```diff
@@ -729,7 +729,7 @@ def to_tensor(self, ret, dtype):
         if hasattr(ret, "shape") and ret.shape:
             ret_type = tl.block_type(dtype, ret.shape)
         else:
-            ret = np.array([ret], dtype=_get_np_dtype(dtype))
+            ret = np.array([ret]).astype(_get_np_dtype(dtype))
```
Comment (pbchekin) [inline python/triton/runtime/interpreter.py:732]: Why is this not reproducible in upstream?

File: python/setup.py [hunk 2]
```diff
@@ -665,8 +665,6 @@ def get_install_requires():
             "autopep8",
             "flake8",
             "isort",
-            # FIXME: pytorch<2.3.0 doesn't support numpy 2.0
-            "numpy<2.0",
```
Comment (pbchekin) [inline python/setup.py:669]: Keep "numpy" as test dependency as it's done in upstream (https://github.com/triton-lang/triton/blob/main/python/setup.py#L661). See also #2053.

Comment (Retribution98) [reply python/setup.py:669]: Sure, I saw, but it is needed to check test passing.
I can discard this changes if test is passed.

File: python/triton/runtime/interpreter.py (hunk 1 above)
Comment (Retribution98) [reply python/triton/runtime/interpreter.py:732]: As far as I can see, they use the pinned numpy version 1.26.4 for ci.

Comment (whitneywhtsang) [reply python/triton/runtime/interpreter.py:732]: Does this change work with 1.24.4 numpy? Can we upstream this change?

Comment (Retribution98) [reply python/triton/runtime/interpreter.py:732]: Yes, upstream uses the same behavior for [z_ref ](https://github.com/triton-lang/triton/blob/cf31278859a170eb2f20407c9f4004cafdf41c0d/python/test/unit/language/test_core.py#L2034)in this test.
Should I try to send this PR upstream?

Comment (whitneywhtsang) [reply python/triton/runtime/interpreter.py:732]: Yes, please upstream this change.

Comment (Retribution98) [reply python/triton/runtime/interpreter.py:732]: @whitneywhtsang 
I've created the [PR](https://github.com/triton-lang/triton/pull/4649)

Comment (pbchekin) [reply python/triton/runtime/interpreter.py:732]: Merged in upstream. Nicely done @Retribution98!

**Comment by Retribution98** [comment]
The [PR](https://github.com/triton-lang/triton/pull/4649) was merged in upstream

**Pull request closed without merging**